src/nodes.coffee

```diff
@@ -4573,6 +4573,57 @@ exports.Switch = class Switch extends Base
     fragments.push @makeCode @tab + '}'
     fragments
 
+  astType: -> 'SwitchStatement'
+
+  casesAst: (o) ->
+    cases = []
+
+    for kase, caseIndex in @cases
+      {conditions: tests, block: consequent} = kase
+      tests = flatten [tests]
+      lastTestIndex = tests.length - 1
+      for test, testIndex in tests
+        testConsequent =
+          if testIndex is lastTestIndex
+            consequent
+          else
+            null
+
+        caseLocationData = test.locationData
+        caseLocationData = mergeLocationData caseLocationData, testConsequent.expressions[testConsequent.expressions.length - 1].locationData if testConsequent?.expressions.length
+        caseLocationData = mergeLocationData caseLocationData, kase.locationData, justLeading: yes if testIndex is 0
+        caseLocationData = mergeLocationData caseLocationData, kase.locationData, justEnding:  yes if testIndex is lastTestIndex
+
+        cases.push new SwitchCase(test, testConsequent, trailing: testIndex is lastTestIndex).withLocationDataFrom locationData: caseLocationData
+
+    if @otherwise?.expressions.length
+      cases.push new SwitchCase(null, @otherwise).withLocationDataFrom @otherwise
+
+    kase.ast(o) for kase in cases
+
+  astProperties: (o) ->
+    return
+      discriminant: @subject?.ast(o) ? null
+      cases: @casesAst o
+
+class SwitchCase extends Base
+  constructor: (@test, @block, {@trailing} = {}) ->
+    super()
+
+  children: ['test', 'block']
+
+  astProperties: (o) ->
+    return
+      test: @test?.ast(o) ? null
+      consequent: @block?.ast(o).body ? []
+      trailing: !!@trailing
```

Comment:
`trailing` is an extension to the Babel-style `SwitchCase` AST. It gets used by the ESLint plugin to be able to do Coffeescript-specific code path analysis on switch statements (basically it sees this and acts like there's an implicit `break` statement, which has code path/control flow implications)